src/sdk/nsfs_glacier_backend/tapecloud.js

```diff
@@ -277,42 +306,102 @@ class TapeCloudGlacierBackend extends GlacierBackend {
     }
 
     /**
      * _recall should perform recall
-     * 
+     *
      * NOTE: Must be overwritten for tests
-     * @param {string} file 
-     * @param {(entry: string) => Promise<void>} recorder 
+     * @param {string} file
+     * @param {(entry: string) => Promise<void>} failure_recorder
+     * @param {(entry: string) => Promise<void>} success_recorder
+	 * @returns {Promise<boolean>}
      */
-    async _recall(file, recorder) {
-        return recall(file, recorder);
+    async _recall(file, failure_recorder, success_recorder) {
+        return TapeCloudUtils.recall(file, failure_recorder, success_recorder);
     }
 
     /**
      * _process_expired should process expired objects
-     * 
+     *
      * NOTE: Must be overwritten for tests
      */
     async _process_expired() {
-        return process_expired();
+        return TapeCloudUtils.process_expired();
+    }
+
+    /**
+     * finalizes the restore by setting the required EAs
+     *
+     * @param {nb.NativeFSContext} fs_context
+     * @param {string} entry_path
+    */
+    async _finalize_restore(fs_context, entry_path) {
+        dbg.log2('TapeCloudGlacierBackend.restore._finalize_restore - entry:', entry_path);
+
+        const entry = new NewlineReaderEntry(fs_context, entry_path);
+        let fh = null;
+        try {
+            fh = await entry.open();
+
+            const stat = await fh.stat(fs_context, {
+                xattr_get_keys: [
+                    GlacierBackend.XATTR_RESTORE_REQUEST,
+                ]
+            });
+
+            const days = Number(stat.xattr[GlacierBackend.XATTR_RESTORE_REQUEST]);
+
+            // In case of invocation on the same file multiple times,
+            // this xattr will not be present hence `days` will be NaN
+            if (isNaN(days)) {
+				dbg.warn("TapeCloudGlacierBackend._finalize_restore: days is NaN - skipping restore for", entry_path);
+				return;
+			}
+
+            const expires_on = GlacierBackend.generate_expiry(
+                new Date(),
+                days,
+                config.NSFS_GLACIER_EXPIRY_TIME_OF_DAY,
+                config.NSFS_GLACIER_EXPIRY_TZ,
+            );
+
+            // First set the expiry so that we don't lose the number of days in
+            // case of a partial failure. `replacexattr` first clears the xattrs
+            // and then proceeds to set the xattr which makes it highly prone to
+            // partial failure such that we lose the attribute forever and
+            // consequently never really process the restore request (until it
+            // is submitted again).
+
+            await fh.replacexattr(fs_context, {
+                [GlacierBackend.XATTR_RESTORE_EXPIRY]: expires_on.toISOString(),
+            });
+
+            await fh.replacexattr(fs_context, undefined, GlacierBackend.XATTR_RESTORE_REQUEST);
+        } catch (error) {
+            dbg.error(`failed to process ${entry.path}`, error);
+			throw error;
```

Comment:
I still see a lot of indentation issues, can you please use a formatter that will fix it?

Reply:
Fixed